test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageViewLocationExpanderTest.cs

```diff
@@ -0,0 +1,144 @@
+
+            // Act
+            expander.PopulateValues(context);
+
+            // Assert
+            Assert.Empty(context.Values);
+        }
+
+        [Fact]
+        public void ExpandLocations_NoOp_ForNonPage()
+        {
+            // Arrange
+            var context = CreateContext(pageName: null);
+            var locations = new string[]
+            {
+                "/ignore-me",
+            };
+
+            var expander = CreateViewLocationExpander();
+
+            // Act
+            var actual = expander.ExpandViewLocations(context, locations);
+
+            // Assert
+            Assert.Equal(locations, actual);
+        }
+
+        [Fact]
+        public void ExpandLocations_NoOp_WhenLocationDoesNotContainPageToken()
+        {
+            // Arrange
+            var context = CreateContext(pageName: null);
+            var locations = new string[]
+            {
+                "/ignore-me",
+            };
+
+            var expander = CreateViewLocationExpander();
+
+            // Act
+            var actual = expander.ExpandViewLocations(context, locations);
+
+            // Assert
+            Assert.Equal(locations, actual);
+        }
+
+        [Theory]
+        [InlineData("/Index", new string[] { "/{0}.cshtml" })]
+        [InlineData("/Edit", new string[] { "/{0}.cshtml" })]
+        [InlineData("/Customers/Add", new string[] { "/Customers/{0}.cshtml", "/{0}.cshtml" })]
+        public void ExpandLocations_ExpandsDirectories_WhenLocationContainsPage(
+            string pageName,
+            string[] expected)
+        {
+            // Arrange
+            var context = CreateContext(pageName: pageName);
+
+            var locations = new string[]
+            {
+                "/{1}/{0}.cshtml",
+            };
+
+            var expander = CreateViewLocationExpander();
+
+            // Act
+            var actual = expander.ExpandViewLocations(context, locations);
+
+            // Assert
+            Assert.Equal(expected, actual.ToArray());
+        }
+
+        public void ExpandLocations_ExpandsDirectories_MultipleLocations()
+        {
+            // Arrange
+            var context = CreateContext(pageName: "/Pages/Customers/Edit");
+
+            var locations = new string[]
+            {
+                "/{1}/{0}.cshtml",
+                "/More/Paths/{1}/{0}.cshtml",
+                "/Views/Shared/{0}.cshtml",
+            };
+
+            var expected = new string[]
+            {
+                "/Pages/Customers/{0}.cshtml",
+                "/Pages/{0}.cshtml",
+                "/More/Paths/Customers/{0}.cshtml",
+                "/More/Paths/{0}.cshtml",
+                "/Views/Shared/{0}.cshtml",
+            };
+
+            var expander = CreateViewLocationExpander();
+
+            // Act
+            var actual = expander.ExpandViewLocations(context, locations);
+
+            // Assert
+            Assert.Equal(expected, actual.ToArray());
+        }
+
```

Comment:
Can we add a test to verify how this works with the `LanguageViewLocationExpander`? It's the only other one we ship in the box afaik.